analyses/sex-prediction-from-RNASeq/04-present_results.Rmd

```diff
@@ -0,0 +1,249 @@
+  number_transcripts <- data.frame(ID=transcript_tail_percents, number_training_transcripts=number_transcripts)
+
+# join predictive accuracy and # training transcript values by ID  
+  pre_final_df <- suppressWarnings(inner_join(predictive_accuracies, number_transcripts, by="ID"))
+
+
+#here, we collect the number of non-zero features from each model coefficients file in the models directory  
+  results_directory <- file.path(params$model_dir)
+  files_to_process <- list.files(path = results_directory, pattern = "_model_coefs.tsv")
+  first_pos <- sapply(files_to_process, function(x) regexpr(params$seed, x))
+  last_pos <- sapply(files_to_process, function(x) regexpr("model_coefs", x))
+  transcript_tail_percents <- substr(files_to_process, 1, last_pos-2)
+  
+  files_to_process_names <- file.path(params$model_dir, files_to_process)
+  number_non_zero <- sapply(files_to_process_names, function(x) 
+    nrow(read.delim(x, header=TRUE, sep="\t", stringsAsFactors = FALSE)))
+
+#store # non-zero features values in a data frame with ID=unique portion of the model coeffs file name  
+  number_non_zero <- data.frame(ID=transcript_tail_percents, number_features=number_non_zero)
+
+#join # non-zero features to predictive accuracies and # training transcript values by ID  
+  final_df <- suppressWarnings(inner_join(pre_final_df, number_non_zero, by="ID"))
+#join DFs two-by-two to suppress 'joining by' messages.  Cannot suppress them with Reduce
+  
+
+#store this data frame in the list of data frames by targetColumn
+  df_list[[t]] <- final_df
+
+  
+}
+
+
+```
+
+### **Overview and Conclusion**
+
+The 01-clean_split_data.R, 02-train_elasticnet.R, 03-evaluate_model.R pipeline trains and evaluates an elasticnet logistic regression model to predict sex from RNASeq data.  
+The training features are gene expression transcripts, and the training labels are reported_gender values for each sample.
+
+The pipeline is a response to issue [#84](https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/84) which was raised to check whether, in some histologies, silencing might be breaking down, potentially resulting in changes in X inactivation. 
+Based on the accuracy achieved here, this is probably not happening, and this classifier can be helpful to predict values for datasets without annotated sex information.
+
+
+
+### **Strength of Calls**
+
+The model generates a probability that a given sample is Female. 
+Female/Male calls are made using a 50% threshold. Predictive accuracy varies by the number of training transcripts used.  
+The plot below shows model probability values, generated by the most accurate set of training transcripts, for all 258 samples in the test set.  
+The test set points are split on the x-axis by reported_gender (ground truth) values.
+Erroneous calls are displayed in red.  
+
+
+```{r echo=FALSE}
+
+#for each value in targetColumns argument, find the index of the row corresponding to the maximum predictive accuracy
```

Comment:
I thought we should have one copy of this plot because we only trained on the reported_gender values.  We can calculate values of maximum_accuracy_index for each target_column value, and use them in this plot and the subsequent plots, but if we are going to vary the model trained on reported_gender values according to test target column, shouldn't we just train on germline_sex_estimate instead when those values are available?

Reply:
The single plot sounds good. I think this is a matter of my interpretation of the comment not matching the implementation/functionality of the code.

Reply:
changed the comment and the logic of the data supporting the plot in the latest push.  Still just one copy of Strength of Calls.

Reply:
To check reproducibility, I looked at model features and coefficients for two separate runs of the pipeline.  They agree.

In the Strength of Calls plot, there are differences in appearance for two separate runs of the pipeline because ggplot varies horizontal placement of points in the dot plot.  For two separate runs of the plot, there is, however, a 1-to-1 correspondence between the y-values.  So, the same result is being reported in the two plots.